File: <100/062/062.jl
function sort_str(s :: String) :: String
  return s |> collect |> sort |> join
end


function solve(upper_limit :: Int) :: Int
  count_memo :: Dict{String, Int} = Dict()
  minimum_memo :: Dict{String, Int} = Dict()
  for i ∈ 346:upper_limit
    cubed = i^3
    s = cubed |> string |> sort_str
    if haskey(count_memo, s)
      count_memo[s] += 1
    else
      count_memo[s] = 1
      minimum_memo[s] = cubed
    end
  end

  good_vals = [minimum_val for (minimum_val, count) ∈ zip(values(minimum_memo), values(count_memo)) if count == 5]
  return minimum(good_vals)
end


function main() :: Nothing
  upper_limit = 10^4
  upper_limit |> solve |> println
end


@time main()

File: <100/062/062.py
def sort_str(s: str) -> str:
    l = sorted(s)
    return "".join(l)


def solve(upper_limit: int) -> int:
    count_memo: dict[str, int] = {}
    minimum_memo: dict[str, int] = {}
    for i in range(346, upper_limit):
        cubed = i**3
        s = str(cubed)
        s = sort_str(s)
        if s in count_memo:
            count_memo[s] += 1
        else:
            count_memo[s] = 1
            minimum_memo[s] = cubed
    
    good_vals = [minimum_val for minimum_val, count in zip(minimum_memo.values(), count_memo.values()) if count == 5]
    return min(good_vals)


def main() -> int:
    upper_limit = 10**5
    answer = solve(upper_limit)
    return answer


if __name__ == "__main__":
    answer = main()
    print(answer)
